### PR TITLE
check if update_db_packages folder is empty before upgrade

### PR DESCRIPTION
**Pull request by abvgedeika**
if folder 'update_db_packages' is empty we can skip _update_db_packages process

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1566,15 +1566,15 @@ def update_db_binary(self, node_list=None):
             node_list = self.nodes
 
         new_scylla_bin = self.params.get('update_db_binary')
-        if new_scylla_bin:
+        if new_scylla_bin and os.listdir(new_scylla_bin):
```
Comment (amoskong) [inline sdcm/cluster.py]: `new_scylla_bin` is a path to binary file, not directory. Why not use `os.path.exist(new_scylla_bin)`?

`os.listdir(new_scylla_bin)` might raise exception:

```
>>> os.listdir(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: coercing to Unicode: need string or buffer, NoneType found
>>> os.listdir('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: [Errno 2] No such file or directory: ''
```

**Comment by amoskong** [comment]
@abvgedeika 

If we really want to test new binary or package, but it doesn't exist we should not skip updating, error should be raised.

Otherwise, we thought we are testing the new package, actually it uses old scylla.

If updating isn't necessary, then the skipping is safe.

**Comment by roydahan** [comment]
I tend to agree with @amoskong here.
If we won't fail the test, one may think it worked... no one actually goes and check the logs :)

File: sdcm/cluster.py (hunk 1 above)
Comment (lmr) [inline sdcm/cluster.py]: Yep, this seems like a typo

**Comment by abvgedeika** [comment]
ok, I got your thoughts. So, we have 3 options:
1) update_db_packages is set and not empty: update os and update with packages
2) update_db_packages is set and empty: update os only
3) update_db_packages not set: no any upgrades

what should be default behaviour in jenkins when we don't set any packages for upgrade: №2 or №3? 
(case when user want to get results from any _relased version for comparison with his packages, for example with 1.7r02 version)

**Comment by roydahan** [comment]
ok, I see what you are saying...
However, the idea for this job is to run it with "your own rpms"...
In order to compare it to one of the releases, one shouldn't run the release test again, but should compare it with the results we got from our "official runs" (those that run in the jobs Amos created - ami-perf-regression...).

Therefore, I would go with the more defensive approach, and if something didn't work with the update (binary or packages), I would fail the job.

**Comment by abvgedeika** [comment]
okay, then we don't need these changes :)

**Pull request closed without merging**